File: julia/linear_regression.jl
using BenchmarkTools;
using DataFrames;
using GLM;


function regression(X::Matrix{Float64}, y::Vector{Float64})
    ols = lm(X, y);
    return ols;
end

randGenerator()::Float64 = rand([1/x for x in 1:100]);
randVec()::Vector{Float64} = [x+(x*randGenerator()) for x in 0:1000000, y in 0:1];

function measure_regression()
    X::Matrix{Float64} = hcat([x+(x*randGenerator()) for x in 0:1000000]);
    y::Vector{Float64} = [x for x in 0:1000000];
    #data::DataFrame = DataFrame(X=X, y=y);
    #data::Vector{Float64, 2} = randVec();
    data = randVec();
    exec_times::Vector{Float64} = zeros(1000);
    for i in 1:length(exec_times)
        println(i);
        result = @timed regression(X, y);
        exec_times[i] = result.time;
    end
    println(median(exec_times));

end


function main()
    measure_regression();
end


main()

File: R/linear_regression.R

regression <- function(data){
  ols = lm(y ~ X, data)
  return(ols)
}

gen_list <- function(){
  v = c(1:1000000)
  for(i in 1:length(v)){
    v[i] = 1/sample(1:100, 1)[1]*v[i] + v[i]
  }
  return(v)
}


main <- function(){

  X = c(1:1000000)
  y = gen_list()
  exec_time_list = c()
  data <- data.frame(
    X = X,
    y = y
  )
  print("Dale")
  
  for(i in 1:1000){
    print(i)
    start_time <- Sys.time()
    f = regression(data)
    end_time <- Sys.time()
    exec_time = end_time - start_time
    exec_time_list = append(exec_time_list, exec_time)
  }
  print(median(exec_time_list))
  
}

